### PR TITLE
Switch event implementation to signals - Closes #186

### LINKED ISSUE #186: Move event triggers to SDK compatibility layer
### Description

Move event triggers to SDK compatibility layer and make different events trigger tasks provided by the core service. 

1. Implement project wide signal support

Move signals to `shared/signals.js`. Early signal implementation to move around:

https://github.com/LiskHQ/lisk-service/blob/65457c1cf0e8122b507a4f0a5ed27351875e3271/services/core/events/blockchain.js#L27

2. Move socket.io references to corresponding SDK compatibility layers (v2/v3/v4). The corresponding socket events can trigger signals and propagate the data further.

### Motivation

- Future SDK versions require version-specific Lisk Core event triggers
- They can be propagated by the signal library which is much more convenient than a fixed socket implementation.

### Acceptance Criteria

- Lisk Core events are propagated to the gateway
- Blockchain events refer to blockchain compatibility layer through the signal library

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?
This PR resolves #186

### How was it solved?

- [x] Signal library is applied
- [x] Events are propagated in the SDKv2
- [x] Events are propagated in the SDKv3
- [x] Events are propagated in the SDKv4
- [x] Add Jenkins tests

### How was it tested?

Manually

### CODE REVIEW

File: services/core/shared/core/delegateUtils.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	const db = await pouchdb(config.db.collections.delegates.name);
+	const dbResult = await db.find({ selector: { address } });
+	if (dbResult.length === 1) return dbResult[0].username;
+	return null;
+};
+
+const getAddressByUsername = async (username) => {
+	const db = await pouchdb(config.db.collections.delegates.name);
+	const dbResult = await db.find({ selector: { username } });
+	if (dbResult.length === 1) return dbResult[0].username;
```
Comment (sameersubudhi) [inline services/core/shared/core/delegateUtils.js:29]: `getAddressByUsername` must return the delegate address.

```suggestion
	if (dbResult.length === 1) return dbResult[0].address;
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Apart from the suggested logical fix, the remaining code changes look good.

**Review by nagdahimanshu — COMMENTED** [review]
Given branch is behind the development, pull the change before merge (Y).

**Review by nagdahimanshu — APPROVED** [review]
Looks good!

**Review by nagdahimanshu — APPROVED** [review]
Looks good!

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**